File: C++/Stack/Stack_02.cpp
////////////////////////////////////////////////////////////
//  
//  Program Name    : Stack Using Singly Linear Linked List
//  Program Number  : 002
//  Description     : This program demonstrates the
//                    implementation of Stack Data Structure
//                    using Singly Linear Linked List.
//
//                    Operations Implemented:
//                    1. Push
//                    2. Pop
//                    3. Peep
//                    4. Display
//                    5. Count
//
//  Concepts Used   : Class, Objects, Dynamic Memory
//                    Allocation, Linked List, Stack (LIFO)
//
//  Input           : Integer Values
//  Output          : Stack Operations Result
//
//  Language        : C++
//  Compiler        : GNU G++
//
//  Author          : Atharv Tushar Bhosale
//  Date            : August 2026
//
////////////////////////////////////////////////////////////

// Stack 

#include<iostream>

using namespace std;

#pragma pack(1)
struct node
{
    int data;
    struct node* next;
};

class Stack
{
    private:
        struct node* first;
        int iCount;

    public:
        Stack();
        void Push(int iNo);             // Insert First
        int Pop();                     // Delete First
        int Peep();                    // Delete First (it will give data only not deleting data)
        void Display();
        int Count();
};

Stack :: Stack()
{
    this -> first = NULL;
    this -> iCount = 0;
}

void Stack :: Push(int iNo)           
{
    struct node* newn = NULL;

    newn = new struct node();
    newn -> data = iNo;
    newn -> next = NULL;

    if(first == NULL)
    {
        first = newn;
    }
    else
    {
        newn -> next = first;
        first = newn;
    }
    iCount++;
}

int Stack :: Pop()
{
    return 0;
}           

int Stack :: Peep()   
{}

void Stack :: Display()
{}

int Stack :: Count()
{
    return iCount;
}

int main()
{
    Stack sobj;
    int iRet = 0;

    sobj.Push(11);
    sobj.Push(21);
    sobj.Push(51);
    sobj.Push(101);

    iRet = sobj.Count();
    cout<<"Elements of this stack:"<<iRet<<endl;

    return 0;
}

////////////////////////////////////////////////////////////
//
//  End of Program
//
//  Stack Using Singly Linear Linked List
//
//  Operations Covered:
//      - Push
//      - Pop
//      - Peep
//      - Display
//      - Count
//
//  Author : Atharv Tushar Bhosale
//  Date   : August 2026
//
////////////////////////////////////////////////////////////

File: Java/Stack/Stack_02.java
/*
===============================================================================
Program Name    : Stack Push Operation
Program Number  : 002
File Name       : Stack_02.java

Description     : This program demonstrates insertion
                  of elements into Stack using
                  Push operation.

Concepts Used:
1. Class and Object
2. Linked List
3. Stack Data Structure
4. Dynamic Memory Allocation

Date            : August 2026
Language        : Java
Author          : Atharv Tushar Bhosale
===============================================================================
*/

class node
{
    public int data;
    public node next;
}

class Stack
{
    private node first;
    private int iCount;

    public Stack()
    {
        first = null;
        iCount = 0;
    }

    ///////////////////////////////////////////////////////////////////////////
    //
    // Function Name : Push
    // Description   : Insert element at First
    //
    ///////////////////////////////////////////////////////////////////////////

    public void Push(int iNo)
    {
        node newn = null;

        newn = new node();

        newn.data = iNo;
        newn.next = null;

        newn.next = first;
        first = newn;

        iCount++;
    }
}

class Stack_02
{
    public static void main(String A[])
    {
        Stack sobj = new Stack();

        sobj.Push(11);
        sobj.Push(21);
        sobj.Push(51);
        sobj.Push(101);

        System.out.println("Elements inserted successfully");
    }
}

/*
===============================================================================
Sample Output

Elements inserted successfully

Stack :

|101|
|51|
|21|
|11|

===============================================================================
End of Program
===============================================================================
*/
